docs/contribute/unit-tests.md

```diff
@@ -1,23 +1,130 @@
+## Testing Workflow
+
+1. Clone the repo using 
+```
+git clone https://github.com/Lind-Project/lind-wasm.git
+```
+2. Change directory to repo 
+```
+cd lind-wasm
+```
+3. Build Docker Image 
+```
+docker build -t testing_image -f .devcontainer/Dockerfile --build-arg DEV_MODE=true --platform=linux/amd64 .
+```
+4. Run the image 
+```
+docker run -it testing_image /bin/bash
+```
+5. Build toolchain (glibc and wasmtime)
+```
+bazel build //:make_glibc //:make_wasmtime
+```
+6. Run the test suite 
+```
+bazel run //:python_tests
+```
+(This will run the whole test suite.  Use `scripts/wasmtestreport.py --help` to
+list available arguments and flags)
+Note: Pass test suite arguments using
+```
+bazel run //:python_tests -- <wasmtestreport arguments>
+```
+For example: 
+```
+bazel run //:python_tests -- --timeout 10
+```
+
+
+
+## What test suite does
+
+1. **Test Case Collection:** Scans `unit-tests` folder for `.c` files.
+
+2. **Filtering:** Applies include/exclude filters (`--run`, `--skip`, and
+   `skip_test_cases.txt`).
+3. **Checking for expected outputs:** Test suite looks for expected output from
+   native run (the output of running the test case after compiling using gcc).
+   If it's not found, we will compile using gcc and run it to get the native
+   output.
+4. **Comparing Outputs:** For deterministic test cases, the outputs are directly
+   compared and considered a success if they match. For non-deterministic test
+   cases, the outputs are parsed and compared using a Python script.
+5. **Result Recording:** All test outcomes are stored with status, error type,
+   and full output.
+6. **Reporting:** JSON and HTML test reports are generated.
+
+
+
+## Output Files
+- **JSON Report:** Detailed test summary in JSON format, saved at `lind-wasm/results.json`
+- **HTML Report:** Human-readable visualization of test outcomes, saved at `lind-wasm/report.html`
+
+The output will show the total number of test cases, along with counts for
+successes, failures, and each of the following error types:
+
+- "Failure_native_compiling": Failed during GCC compiling
+- "Failure_native_running": Failed while running GCC compiled binary
+- "Native_Segmentation_Fault": Segmentation Fault while running GCC binary
+- "Native_Timeout": Timed Out during GCC run
+- "Lind_wasm_compiling": Failed during compilation using lind-wasm
+- "Lind_wasm_runtime": Failed while running lind-wasm compiled binary
+- "Lind_wasm_Segmentation_Fault": Segmentation Fault while running wasm binary
+- "Lind_wasm_Timeout": Timed out During Lind Wasm run
+- "Output_mismatch": Mismatch in GCC and Wasm outputs
+- "Unknown_Failure": Unknown Failure
+
+The report also includes individual test case results, showing status
+(success/failure), error type (if any), and output generated.
+
+The outputs are split into deterministic and non-deterministic based on how the
+lind-wasm outputs are compared to the native gcc output. 
```

Comment:
Same here. I think the difference between deterministic and non-deterministic is already explained elsewhere.

Reply:
I agree they're explained elsewhere, but in this case, I don't see a harm in a brief mention of it in this context.  That said, removing it is trivial if that's the consensus.

Reply:
Let's leave it then. 

What about the other suggestions from https://github.com/Lind-Project/lind-wasm/pull/215#pullrequestreview-2899448730?